File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = {};
        
        int l = 0;
        int r = 0;
        int sum = 0;
        int size = sequence.length;
        int ans1 = 0;
        int ans2 = 0;
        
        
        for(r = 0; r < sequence.length; r++){
            sum += sequence[r];
            
            if(sum > k){
                while(sum>k){
                sum -= sequence[l];
                    l++;
        }
            }
                if(sum == k){
                    if(size > r-l){
                        size = r-l;
                        ans1 = l;
                        ans2 = r;
                        
                    }
                    
                    else if(size == r - l){
                        ans1 = Math.min(ans1, l);
                        ans2 = Math.min(ans2, r);
                    }
                }
            }
        answer = new int[]{ans1, ans2};
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    l = r = 0
    answer = [0, len(sequence)]
    sum = sequence[0]
    while True:
        if sum < k:
            r += 1
            if r == len(sequence):
                break
            sum += sequence[r]
        else:
            if sum == k:
                if r-l < answer[1] -answer[0]:
                    answer = [l, r]
            sum -= sequence[l]
            l+=1
    return answer